app/Models/Camp.php

```diff
@@ -21,7 +19,132 @@ public static function campLink($topicNum,$campNum,$title,$campName){
+        return self::select('topic.topic_name', 'topic.namespace_id', 'camp.*', 'namespace.name as namespace_name', 'namespace.name')
+            ->join('topic', 'topic.topic_num', '=', 'camp.topic_num')
+            ->join('namespace', 'topic.namespace_id', '=', 'namespace.id')
+            ->where('camp.topic_num', $filter['topicNum'])->where('camp_name', '=', 'Agreement')
+            ->where('camp.objector_nick_id', '=', NULL)
+            ->where('topic.objector_nick_id', '=', NULL)
+            ->where('topic.go_live_time', '<=', $asofdate)
+            ->latest('topic.go_live_time')->first();
+    }
+
+    public static function getLiveCamp($filter = array())
+    {
+        $filterName = $filter['asOf'];
+        if (!$filterName) {
+            $filter['asOf'] = 'default';
+        }
+        return self::LiveCampasOfFilter($filter);
+    }
+
+    private function LiveCampasOfFIlter($filter)
+    {
+        $asOfFilter = [
+            'default' => self::LiveCampdefaultAsOfFilter($filter),
+            'review'  => self::LiveCampreviewAsofFilter($filter),
+            'bydate'  => self::LiveCampbyDateFilter($filter),
+        ];
+        return $asOfFilter[$filter['asOf']];
+    }
+
+    public static function LiveCampdefaultAsOfFilter($filter)
+    {
+        return self::where('topic_num', $filter['topicNum'])
+            ->where('camp_num', '=', $filter['campNum'])
+            ->where('objector_nick_id', '=', NULL)
+            ->where('go_live_time', '<=', time())
+            ->latest('submit_time')->first();
+    }
+
+    public static function LiveCampreviewAsofFilter($filter)
+    {
+        return self::where('topic_num', $filter['topicNum'])
+            ->where('camp_num', '=', $filter['campNum'])
+            ->where('objector_nick_id', '=', NULL)
+            ->latest('submit_time')->first();
+    }
+
+    public static function LiveCampbyDateFilter($filter)
+    {
+        $asofdate = strtotime(date('Y-m-d H:i:s', strtotime($filter['asOfDate'])));
+        if (isset($filter['nofilter']) && $filter['nofilter']) {
+            $asofdate  = time();
+        }
+        return self::where('topic_num', $filter['topicNum'])
+            ->where('camp_num', '=', $filter['campNum'])
+            ->where('objector_nick_id', '=', NULL)
+            ->where('go_live_time', '<=', $asofdate)
+            ->latest('submit_time')->first();
+    }
+
+    public function scopeCampNameWithAncestors($query, $camp, $campname = '', $title = '', $filter = array(), $breadcrum = false)
+    {
+        $as_of_time = time();
+        if (isset($filter['asOf']) && $filter['asOf'] == 'bydate') {
+            $as_of_time = strtotime($filter['asOfDate']);
+        }
+        if (!empty($camp)) {
+            if ($campname != '') {
+                $campname =  ($camp->camp_name) . '/ ' . ($campname);
+            } else {
+                $campname =  ($camp->camp_name);
+            }
+            if (isset($camp) && $camp->parent_camp_num) {
```

Comment:
In the above `if` statement, the $camp is not empty then only the program enters in this. Can we get rid of `isset($camp)` ?